mne_bids_pipeline/_run.py

```diff
@@ -68,18 +69,22 @@ def wrapper(*args, **kwargs):
                 log_info["success"] = False
                 log_info["error_message"] = str(e)
 
+                # Find the limit / step where the error occurred
+                step_dir = pathlib.Path(__file__).parent / "steps"
+                tb = traceback.extract_tb(e.__traceback__)
+                for fi, frame in enumerate(inspect.stack()):
+                    is_step = pathlib.Path(frame.filename).parent.parent == step_dir
+                    del frame
+                    if is_step:
+                        # omit everything before the "step" dir, which will
+                        # generally be stuff from this file and joblib
+                        tb = tb[-fi:]
+                        break
+                tb = "".join(traceback.format_list(tb))
```

Comment:
And this one gives much nicer tracebacks by omitting the `_run.py` and `joblib` stuff, so we get this instead of a 5- or 6-level traceback (here where I've added a `raise RuntimeError` to `_02_find_empty_room`):
```
[15:13:59] │ ❌ init/_02_find_empty_room sub-01 ses-meg run-02 A critical error occurred. The error message was: 

Aborting pipeline run. The full traceback is:

  File "/home/larsoner/python/mne-bids-pipeline/mne_bids_pipeline/steps/init/_02_find_empty_room.py", line 70, in find_empty_room
    raise RuntimeError
```